sql/src/test/java/io/crate/integrationtests/QueryThenFetchIntegrationTest.java

```diff
@@ -87,11 +87,41 @@ public void testPushBasedQTF() throws Exception {
         execute("insert into t (x) values ('a')");
         execute("refresh table t");
 
-        execute("select * from t limit ?", new Object[] {Constants.PAGE_SIZE + 10000 });
+        execute("select * from t limit ?", new Object[]{Constants.PAGE_SIZE + 10000});
         assertThat(response.rowCount(), is(1L));
     }
 
     @Test
+    public void testPushBasedQTFWithPaging() throws Exception {
+        // insert more docs than PAGE_SIZE and query at least 2 times of it to trigger push of
+        // at least 2 pages
+        int docCount = (Constants.PAGE_SIZE * 2) + 2;
```

Comment:
I think the PAGE_SIZE was increased to 1mio,  does this test still run in an acceptable time?